userspace/libsinsp/container.h

```diff
@@ -171,20 +171,41 @@ class sinsp_container_engine_docker
 
 	bool resolve(sinsp_container_manager* manager, sinsp_threadinfo* tinfo, bool query_os_for_missing_info);
 	static void cleanup();
+	static void refresh();
 	static void set_query_image_info(bool query_image_info);
+
+	struct docker_request_info
```

Comment:
Instead of directly providing the url as a part of the request struct, maybe you could specify the path (images vs. containers), api version, and container id separately, and assemble the url in start_docker_request(). I think this would avoid the need for id_to_query_for_ip, as a second query to look up a container resulting from a NetworkMode -> "container:" would just end up looking like any other request.

Reply:
this sounds definitely better, will do